recipes/inkscape/wrapper.c

```diff
@@ -0,0 +1,262 @@
+/*
+    Wrapper for Windows to avoid using bat files
+
+    To build/rebuild with mingw32, do this in the setuptools project directory:
+
+       gcc -DGUI=0           -mno-cygwin -O -s -o setuptools/cli.exe launcher.c
+       gcc -DGUI=1 -mwindows -mno-cygwin -O -s -o setuptools/gui.exe launcher.c
+*/
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <windows.h>
+#include <tchar.h>
+#include <fcntl.h>
+#include <process.h>
+
+
+// http://www.catch22.net/tuts/reducing-executable-size
+// Favour small code
+#pragma optimize("gsy", on)
+
+
+/*if you need it: here is a way to get specific path for different arch... */
+#if __SIZEOF_POINTER__ == 8
+# define ARCH "x64"
+#else
+# define ARCH "i386"
+#endif
+
+/*
+Change these to the *relative* paths your real executable is in
+In the end this is used as
+   <path env><RELATIVE_PATH><wrapper name><EXEC_EXTENSION>
+Slashes are need to be included here!
+<path env> really means one dir below the dir where the wrapper is. Which is
+true for <env>\Scripts and <env>\bin
+*/
+#define RELATIVE_PATH   "\\Library\\inkscape\\"
+#define EXEC_EXTENSION  ".com"
+#define DEBUG 0
+
+
+/* Only change soemthing below where the path is set in the end.. */
+
+int child_pid=0;
+
+
+int fail(char *format, char *data)
+{
+    /* Print error message to stderr and return 2 */
+    fprintf(stderr, format, data);
+    return 2;
+}
+
+
+char *quoted(char *data)
+{
+    int i, ln = strlen(data), nb;
```

Comment:
The return value from `strlen` is `size_t`, not `int`. You would also need to change `i` to avoid signed/unsigned comparison. But don't change `nb` because you run it backwards and check for negatives.